### ISSUE TITLE
Custom Forms List Display

### ISSUE THREAD

**Issue by Fidel-C** [piccolo-orm/piccolo_admin#298]
If a custom form field is set to a list, it displays an add icon button which seems not to function as it should.

Here is my code for instance

```python
class EmailForm(BaseModel):
    subject: str = 'Booking'
    sender: str = 'Booking'
    receivers: list[EmailStr]
    body: str
    
async def emailform_endpoint(request, data: EmailForm):
    message=MessageSchema(
        subject=data.subject,
        subtype=MultipartSubtypeEnum.mixed,
        recipients=data.receivers,
        body=data.body
    )
```

The add icon button tied to the receivers field does not display an email input when clicked. It probably just appends an empty string to the list.

**Comment by sinisaos** [comment]
@Fidel-C Yes, you're right. It's a small fix. I will make PR for that.

**Comment by Fidel-C** [comment]
_Thanks. Looking forward to the fix._

**Comment by dantownsend** [comment]
OK, I see the problem. I put this simple app together which replicates the problem:

```python
from pydantic import BaseModel
from piccolo_admin.endpoints import create_admin, FormConfig
from piccolo.apps.user.tables import BaseUser
from piccolo_api.session_auth.tables import SessionsBase
from piccolo.engine.sqlite import SQLiteEngine
from starlette.requests import Request


DB = SQLiteEngine()


class Model(BaseModel):
    items: list[str]


async def emailform_endpoint(request: Request, data: Model) -> str:
    return 'ok'


class BaseUserCustom(BaseUser, db=DB):
    pass


class SessionsBaseCustom(SessionsBase, db=DB):
    pass


app = create_admin(
    tables=[],
    forms=[
        FormConfig(
            name='Demo',
            pydantic_model=Model,
            endpoint=emailform_endpoint
        )
    ],
    auth_table=BaseUserCustom,
    session_table=SessionsBaseCustom
)

async def main():
    await BaseUserCustom.create_table(if_not_exists=True)

    if not await BaseUserCustom.exists().where(
        BaseUserCustom.username == 'piccolo'
    ):
        await BaseUserCustom.create_user(
            username='piccolo',
            password='piccolo123',
            admin=True,
            active=True
        )

    await SessionsBaseCustom.create_table(if_not_exists=True)


if __name__ == '__main__':
    import asyncio
    asyncio.run(main())

    import uvicorn
    uvicorn.run(app)
```

<img width="707" alt="Screenshot 2023-04-26 at 22 41 21" src="https://user-images.githubusercontent.com/350976/234709295-6def2266-2cf5-4665-a449-7dfa3bc726ca.png">

**Comment by dantownsend** [comment]
The solution suggested by @sinisaos works - so have merged that in 👍

**Comment by dantownsend** [comment]
Should be fixed in `piccolo_admin==0.48.0`.
